manta-crypto/src/arkworks/glv/mod.rs

```diff
@@ -0,0 +1,324 @@
+                .into_repr()
+                .to_bits_be(),
+            p1.into_projective(),
+            p2.into_projective(),
+        )
+    }
+
+    /// Executes a simulatenous multiple point multiplication without windowing.
+    #[inline]
+    fn simultaneous_multiple_point_multiplication(
+        u: Vec<bool>,
+        v: Vec<bool>,
+        p: C::Projective,
+        q: C::Projective,
+    ) -> C {
+        let table = vec![C::zero().into_projective(), p, q, p + q];
+        let mut r = C::zero().into_projective();
+        for i in 0..u.len() {
+            r.double_in_place();
+            r += table[u[i] as usize + 2 * (v[i] as usize)]
+        }
+        r.into_affine()
+    }
+
+    /// Multiplies `point` by `scalar` using the GLV method.
+    #[inline]
+    pub fn scalar_mul<M>(&self, point: &C, scalar: &C::ScalarField) -> C
+    where
+        C: HasGLV<M>,
+    {
+        let (k1, k2, p1, p2) = self.scalars_and_points(point, scalar);
+        Self::simultaneous_multiple_point_multiplication(k1, k2, p1, p2)
+    }
+}
+
+/// HasGLV Trait
+pub trait HasGLV<M>: AffineCurveExt {
+    /// Generates [`GLVParameters`] from some precomputed parameters encoded
+    /// in the marker type `M`.
+    fn glv_parameters() -> GLVParameters<Self>;
+
+    /// Applies the GLV endomorphism to `self`.
+    #[inline]
+    fn glv_endomorphism(&self, beta: &Self::BaseField) -> Self {
+        Self::from_xy_unchecked(*self.x() * beta, *self.y())
+    }
+}
+
+#[cfg(feature = "ark-bls12-381")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "ark-bls12-381")))]
+impl HasGLV<bls12_381::Parameters> for bls12_381::G1Affine {
+    #[inline]
+    fn glv_parameters() -> GLVParameters<Self> {
+        parse_glv_parameters::<bls12_381::G1Affine>(include!("precomputed_glv_values/bls_values"))
+    }
+}
+
+#[cfg(feature = "ark-bn254")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "ark-bn254")))]
+impl HasGLV<bn254::Parameters> for bn254::G1Affine {
+    #[inline]
+    fn glv_parameters() -> GLVParameters<Self> {
+        parse_glv_parameters::<bn254::G1Affine>(include!("precomputed_glv_values/bn_values"))
+    }
+}
+
+/// Parses the GLV parameters from strings.
+#[inline]
+pub fn parse_glv_parameters<C>(glv_parameters: [&str; 5]) -> GLVParameters<C>
+where
+    C: AffineCurveExt,
+{
+    let beta = C::BaseField::from_random_bytes(
+        &glv_parameters[0].parse::<BigUint>().unwrap().to_bytes_le(),
+    )
+    .unwrap();
+    let base_v1 = (
+        BigInt::from_str(glv_parameters[1]).unwrap(),
+        BigInt::from_str(glv_parameters[2]).unwrap(),
+    );
+    let base_v2 = (
+        BigInt::from_str(glv_parameters[3]).unwrap(),
+        BigInt::from_str(glv_parameters[4]).unwrap(),
+    );
+    GLVParameters::<C>::new_unchecked(beta, base_v1, base_v2)
+}
+
+/// Testing Suite
+#[cfg(test)]
+pub mod test {
+    use super::*;
+    use crate::rand::RngCore;
+    use ark_ff::UniformRand;
+    use rand_core::OsRng;
```

Comment:
Use the crate-local definition once the dependency has been re-exported:

```suggestion
    use crate::rand::OsRng;
```